R/preprocess.R

```diff
@@ -1,69 +1,88 @@
-#' Add columns for interval censoring of primary and secondary events
+#' Prepare date data in the `epidist_linelist` format
 #'
-#' @param linelist ...
-#' @param ptime_lwr ...
-#' @param ptime_upr ...
-#' @param pwindow ...
-#' @param stime_lwr ...
-#' @param stime_upr ...
-#' @param swindow ...
+#' @param data A `data.frame` containing line list data
+#' @param pdate_lwr,pdate_upr,sdate_lwr,sdate_upr Strings giving the column of
+#' `data` containing the primary and secondary event upper and lower bounds.
+#' These columns of `data` must be as datetime.
+#' @param obs_date A string giving the column of `data` containing the
+#' observation time as a datetime.
 #' @family preprocess
-#' @autoglobal
 #' @export
-add_event_vars <- function(
-  linelist, ptime_lwr = NULL, ptime_upr = NULL, pwindow = NULL,
-  stime_lwr = NULL, stime_upr = NULL, swindow = NULL
+as_epidist_linelist <- function(
+  data, pdate_lwr = NULL, pdate_upr = NULL, sdate_lwr = NULL, sdate_upr = NULL,
+  obs_date = NULL
 ) {
-  linelist <- .rename_column(linelist, "ptime_lwr", ptime_lwr)
-  linelist <- .rename_column(linelist, "ptime_upr", ptime_upr)
-  linelist <- .rename_column(linelist, "stime_lwr", stime_lwr)
-  linelist <- .rename_column(linelist, "stime_upr", stime_upr)
-  linelist <- .rename_column(linelist, "pwindow", pwindow)
-  linelist <- .rename_column(linelist, "swindow", swindow)
+  class(data) <- c("epidist_linelist", class(data))
 
-  if (is.numeric(pwindow)) {
-    cli::cli_warn("Overwriting using numeric value(s) of pwindow provided!")
-    linelist$pwindow <- pwindow
-  }
+  # Rename columns to our internal names: inefficient and needs refactor
```

Comment:
This should be an issue not a comment

Reply:
Agree

Reply:
https://github.com/epinowcast/epidist/issues/417